### PR TITLE
create /api/v1/beers endpoint

### PR DESCRIPTION
**Pull request by Shiggiddie**
Addresses #20 

NOTE: must create a `secret.js` file under the `app/` directory that contains an object with a `bdb_key` attribute whose value is the brewerydb API key.  Be sure to export the aforementioned object in `secret.js`'s `module.exports` object.
- Creates `/api/v1/beers` endpoint
- Expects a POST payload with a `beer` payload key whose value is the beer desired to be searched
- Returns all of the data returned by brewerydb's beer search API endpoint

### CODE REVIEW

**Comment by Shiggiddie** [comment]
Example usage:

```
curl -d '{"beer":"312 Urban Wheat Ale"}' -H "Content-Type: application/json" http://127.0.0.1:3000/api/v1/beers
{"id":"EaoR0H","name":"312 Urban Wheat Ale","description":"Inspired by the city of Chicago and densely populated with flavor, 312’s spicy aroma of Cascade hops is followed by a crisp, fruity ale flavor delivered in a smooth, creamy body that's immensely refreshing.","abv":"4.2","ibu":"20","glasswareId":9,"srmId":3,"availableId":1,"styleId":112,"isOrganic":"N","labels":{"icon":"https://s3.amazonaws.com/brewerydbapi/beer/EaoR0H/upload_NG15x9-icon.png","medium":"https://s3.amazonaws.com/brewerydbapi/beer/EaoR0H/upload_NG15x9-medium.png","large":"https://s3.amazonaws.com/brewerydbapi/beer/EaoR0H/upload_NG15x9-large.png"},"status":"verified","statusDisplay":"Verified","createDate":"2012-01-03 02:42:37","updateDate":"2015-01-14 01:48:45","glass":{"id":9,"name":"Weizen","createDate":"2012-01-03 02:41:33"},"srm":{"id":3,"name":"3","hex":"FFCA5A"},"available":{"id":1,"name":"Year Round","description":"Available year round as a staple beer."},"style":{"id":112,"categoryId":11,"category":{"id":11,"name":"Hybrid/mixed Beer","createDate":"2012-03-21 20:06:46"},"name":"Light American Wheat Ale or Lager with Yeast","description":"This beer can be made using either ale or lager yeast. It can be brewed with 30 to 75 percent wheat malt, and hop rates may be low to medium. Hop characters may be light to moderate in bitterness, flavor and aroma. Fruity-estery aroma and flavor are typical but at low levels however, phenolic, clove-like characteristics should not be perceived. Color is usually straw to light amber, and the body should be light to medium in character. Diacetyl should not be perceived. Because this style is served with yeast the character should portray a full yeasty mouthfeel and appear hazy to very cloudy. Chill haze is also acceptable. Yeast flavor and aroma should be low to medium but not overpowering the balance and character of malt and hops. These beers are typically served with the yeast in the bottle, and are cloudy when served.","ibuMin":"10","ibuMax":"35","abvMin":"3.5","abvMax":"5.5","srmMin":"4","srmMax":"10","ogMin":"1.036","fgMin":"1.006","fgMax":"1.018","createDate":"2012-03-21 20:06:46"},"type":"beer"}
```

**Comment by Shiggiddie** [comment]
Now also includes a mock response:

```
curl -d '{"beer":"foo","mock":"true"}' -H "Content-Type: application/json" http://127.0.0.1:3000/api/v1/beers
{"id":"EaoR0H","name":"312 Urban Wheat Ale","description":"Inspired by the city of Chicago and densely populated with flavor, 312’s spicy aroma of Cascade hops is followed by a crisp, fruity ale flavor delivered in a smooth, creamy body that's immensely refreshing.","abv":"4.2","ibu":"20","glasswareId":9,"srmId":3,"availableId":1,"styleId":112,"isOrganic":"N","labels":{"icon":"https://s3.amazonaws.com/brewerydbapi/beer/EaoR0H/upload_NG15x9-icon.png","medium":"https://s3.amazonaws.com/brewerydbapi/beer/EaoR0H/upload_NG15x9-medium.png","large":"https://s3.amazonaws.com/brewerydbapi/beer/EaoR0H/upload_NG15x9-large.png"},"status":"verified","statusDisplay":"Verified","createDate":"2012-01-03 02:42:37","updateDate":"2015-01-14 01:48:45","glass":{"id":9,"name":"Weizen","createDate":"2012-01-03 02:41:33"},"srm":{"id":3,"name":"3","hex":"FFCA5A"},"available":{"id":1,"name":"Year Round","description":"Available year round as a staple beer."},"style":{"id":112,"categoryId":11,"category":{"id":11,"name":"Hybrid/mixed Beer","createDate":"2012-03-21 20:06:46"},"name":"Light American Wheat Ale or Lager with Yeast","description":"This beer can be made using either ale or lager yeast. It can be brewed with 30 to 75 percent wheat malt, and hop rates may be low to medium. Hop characters may be light to moderate in bitterness, flavor and aroma. Fruity-estery aroma and flavor are typical but at low levels however, phenolic, clove-like characteristics should not be perceived. Color is usually straw to light amber, and the body should be light to medium in character. Diacetyl should not be perceived. Because this style is served with yeast the character should portray a full yeasty mouthfeel and appear hazy to very cloudy. Chill haze is also acceptable. Yeast flavor and aroma should be low to medium but not overpowering the balance and character of malt and hops. These beers are typically served with the yeast in the bottle, and are cloudy when served.","ibuMin":"10","ibuMax":"35","abvMin":"3.5","abvMax":"5.5","srmMin":"4","srmMax":"10","ogMin":"1.036","fgMin":"1.006","fgMax":"1.018","createDate":"2012-03-21 20:06:46"},"type":"beer"}
```

File: app/app.js [hunk 1]
```diff
@@ -20,12 +23,51 @@ app.use(bodyParser.json());
...
+//app.use(express.bodyParser()); need?
 
 app.get('/', function(req, res, next) {
     console.log('routing to index');
     res.render('index');
     });
 
+app.post('/api/v1/beers', function(req, res, next) {
+    // Construct the brewerydb endpoint to hit
+    endpoint = 'http://api.brewerydb.com/v2/search?q=' + encodeURIComponent(req.body.beer) + '&key=' + secrets.bdb_key
```
Comment (DLavin23) [inline app/app.js]: We do not access `endpoint` until we're already inside of the non-mock response, let's move that there.

File: app/app.js [hunk 2]
```diff
@@ -20,12 +23,51 @@ app.use(bodyParser.json());
...
+        // Hit the endpoint
+        request(endpoint, function (error, response, body) {
+            if (!error && response.statusCode == 200) {
+                var data = JSON.parse(body).data;
+                var matches = 0
+
+                // Create an array that filters the return data such that the last
+                // entry in the array will be the data entry that had the most words
+                // that matched the original queried beer
+                var qualifiers = data.filter( function(el) {
```
Comment (DLavin23) [inline app/app.js:49]: This current filtering mechanism has potential to be wasteful, let's clean it up with a `break` algorithm instead.

Comment (Shiggiddie) [inline app/app.js:49]: I'll clean up this algorithm in a new PR so SPA work can begin, cool?

Comment (DLavin23) [inline app/app.js:49]: :+1: yes

Comment (anthony-dandrea) [inline app/app.js:49]: Hurry up :gun:

**Pull request merged**